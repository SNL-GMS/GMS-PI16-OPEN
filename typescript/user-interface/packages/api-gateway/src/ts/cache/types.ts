import { UserActionDescription } from '@gms/common-model/lib/cache/types';
import Immutable from 'immutable';

/**
 * Specifies the type (method) for committing to the Global Cache.
 */
export type CommitFunc<T> = (items: T[]) => void;

/**
 * User Cache Processor interface.
 * Defines all of the common method definitions required
 * for a user cache item.
 */
// eslint-disable-next-line @typescript-eslint/naming-convention
export interface IUserCacheItem<T> {
  has(id: string): boolean;
  get(id: string): T;
  getAll(): T[];
  set(item: T): void;
  setAll(items: T[]): void;
  remove(item: T): void;
  removeAll(item: T[]): void;
  commitAll(): void;
  commitWithIds(ids: string[]): void;
  updateFromGlobalCache(cachedEntries: T[], overwrite: boolean): void;
}

/**
 * User Cache interface.
 * Defines all of the common method definitions required for a user cache.
 */
// eslint-disable-next-line @typescript-eslint/naming-convention
export interface IUserCache {
  getRedoPriorityOrder(id: string): number | undefined;
  getHistory(): History[];
  clearHistory(): void;
}

/**
 * User context for apollo and user sessions
 */
export interface UserContext {
  readonly sessionId: string;
  readonly userName: string;
  readonly userRole: string;
}

/**
 * Defines the interface of User Action.
 */
export interface UserAction {
  /**
   * The user action description
   */
  readonly description: UserActionDescription;

  /**
   * Returns the toString value of a user action.
   *
   * @returns the string representation of the action
   */
  toString(): string;
}

/**
 * Defines the hypothesis type for a hypothesis change.
 */
export enum HypothesisType {
  EventHypothesis = 'EventHypothesis',
  SignalDetectionHypothesis = 'SignalDetectionHypothesis'
}

/**
 * Represents hypothesis information for a change.
 * Defines a mapping from a hypothesis id to the main
 * object id.
 */
export interface HypothesisChangeInformation {
  readonly id: string;
  readonly hypothesisId: string;
  readonly type: HypothesisType;
  readonly parentId: string;
  readonly userAction: UserAction;
}

/**
 * Represents a history change.
 */
export interface HistoryChange {
  readonly id: string;
  readonly active: boolean;
  readonly eventId: string;
  readonly conflictCreated: boolean;
  readonly hypothesisChangeInformation: HypothesisChangeInformation;
}

/**
 * Represents the history.
 */
export interface History {
  readonly id: string;
  readonly description: UserActionDescription;
  readonly changes: Immutable.List<HistoryChange>;
}

/**
 * Invalid data; typically data that has been deleted.
 */
export interface InvalidData {
  readonly eventIds: string[];
  readonly signalDetectionIds: string[];
}
