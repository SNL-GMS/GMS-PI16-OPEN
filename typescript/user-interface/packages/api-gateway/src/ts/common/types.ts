import { CommonTypes } from '@gms/common-model';

/**
 * Common model definitions shared across gateway data APIs
 */

/**
 * Processing Context part of data structure that filtered waveform
 * control service uses to tell who/where the request came from. In our case
 * from Interactive UI user
 */
export interface ProcessingContext {
  readonly analystActionReference: AnalystActionReference;
  readonly processingStepReference: ProcessingStepReference;
  readonly storageVisibility: string;
}

/**
 * Analyst action reference
 */
export interface AnalystActionReference {
  readonly processingStageIntervalId: string;
  readonly processingActivityIntervalId: string;
  readonly analystId: string;
}

/**
 * Processing step reference
 */
export interface ProcessingStepReference {
  readonly processingStageIntervalId: string;
  readonly processingSequenceIntervalId: string;
  readonly processingStepId: string;
}

/**
 * Test data paths used when reading in data
 */
export interface TestDataPaths {
  readonly dataHome: string;
  readonly jsonHome: string;
  readonly additionalDataHome: string;
  readonly integrationDataHome: string;
}

/**
 * Represents calibration information associated with a waveform
 */
export interface ProcessingCalibration {
  readonly factor: number;
  readonly factorError: number;
  readonly period: number;
  readonly timeShift: number;
}

/**
 * Represents the configured type of data source the API Gateway provides access to - values:
 * Local - The API gateway loads data from local file storage for testing purposes
 * Service - The API gateway uses services to provide access to backend (e.g. OSD) data
 */
export enum AccessorDataSource {
  Local = 'Local',
  Service = 'Service'
}

/**
 * Enumerated list of source types used to compute distances to
 */
export enum DistanceSourceType {
  Event = 'Event',
  UserDefined = 'UserDefined'
}

/**
 * Represents input arguments for calculating distance measurement
 * relative to a specified source location
 */
export interface DistanceToSourceInput {
  // The type of the source the distance is measured to (e.g. and event)
  readonly sourceType: DistanceSourceType;

  // the unique ID of the source object
  readonly sourceId: string;
}

/**
 * Client log, object that describes the client log message
 */
export interface ClientLog {
  readonly logLevel: CommonTypes.LogLevel;
  readonly message: string;
  readonly time?: string;
}
