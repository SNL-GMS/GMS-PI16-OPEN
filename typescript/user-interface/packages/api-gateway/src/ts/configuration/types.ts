import { ConfigurationTypes } from '@gms/common-model';
import Immutable from 'immutable';

/**
 * Overall configuration
 */
export interface Configuration {
  /**
   * AnalystConfiguration keyed on user role
   */
  analystConfiguration: Immutable.Map<string, ConfigurationTypes.ProcessingAnalystConfiguration>;

  /**
   * SohConfigurationOld
   */
  sohConfiguration: ConfigurationTypes.SohConfiguration;

  /**
   * CommonConfiguration keyed on user role
   */
  commonConfiguration: Immutable.Map<string, ConfigurationTypes.ProcessingCommonConfiguration>;
}

/**
 * Default user role
 */
export const defaultUserRole = 'DEFAULT';
