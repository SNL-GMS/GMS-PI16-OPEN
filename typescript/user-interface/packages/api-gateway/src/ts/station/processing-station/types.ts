import { ChannelTypes, CommonTypes, ConfigurationTypes } from '@gms/common-model';

/**
 * Model definitions for the processing
 * StationGroup, Station, ChannelGroup and Channel data API
 */

/**
 * Encapsulates station-related data cached in memory
 */
export interface ProcessingStationData {
  stationGroupMap: Map<string, ProcessingStationGroup>;
  stationMap: Map<string, ProcessingStation>;
  channelGroupMap: Map<string, ProcessingChannelGroup>;
  channelMap: Map<string, ProcessingChannel>;
  sohStationGroupNameMap: Map<string, ConfigurationTypes.SOHStationGroupNameWithPriority[]>;
}
/**
 * Represents a group of stations used for monitoring.
 * This is the processing equivalent of the ReferenceNetwork.
 */
export interface ProcessingStationGroup {
  name: string;
  description: string;
  stations: ProcessingStation[];
}

/**
 * Represents an installation of monitoring sensors for the purposes of processing.
 * Multiple sensors can be installed at the same station.
 */
export interface ProcessingStation {
  name: string;
  type: CommonTypes.StationType;
  description: string;
  relativePositionsByChannel: Map<string, CommonTypes.Position>;
  location: CommonTypes.Location;
  channelGroups: ProcessingChannelGroup[];
  channels: ProcessingChannel[];
}

/**
 * ChannelGroupType enum represents the different groupings of channels
 */
export enum ChannelGroupType {
  PROCESSING_GROUP = 'PROCESSING_GROUP',
  SITE_GROUP = 'SITE_GROUP'
}

/**
 * Represents a physical installation (e.g., building, underground vault, borehole)
 * containing a collection of Instruments that produce Raw Channel waveform data.
 */
export interface ProcessingChannelGroup {
  name: string;
  description: string;
  location: CommonTypes.Location;
  type: ChannelGroupType;
  channels: ProcessingChannel[];
}

/**
 * Represents a source for unprocessed (raw) or processed (derived) time series data
 * from a seismic, hydroacoustic, or infrasonic sensor.
 */
export interface ProcessingChannel {
  name: string;
  canonicalName: string;
  description: string;
  station: string;
  channelDataType: ChannelTypes.ChannelDataType;
  channelBandType: ChannelTypes.ChannelBandType;
  channelInstrumentType: ChannelTypes.ChannelInstrumentType;
  channelOrientationType: ChannelTypes.ChannelOrientationType;
  channelOrientationCode: string;
  units: CommonTypes.Units;
  nominalSampleRateHz: number;
  location: CommonTypes.Location;
  orientationAngles: Orientation;
  configuredInputs: string[];
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  processingDefinition: Map<string, any>;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  processingMetadata: Map<ChannelProcessingMetadataType, any>;
}

/**
 * Represents the orientation angles used in processing channels
 */
export interface Orientation {
  horizontalAngleDeg: number;
  verticalAngleDeg: number;
}

/**
 * Represents the type of processing metadata values that can appear as keys in the
 */
export enum ChannelProcessingMetadataType {
  // General properties
  CHANNEL_GROUP = 'CHANNEL_GROUP',

  // Filtering properties
  FILTER_CAUSALITY = 'FILTER_CAUSALITY',
  FILTER_GROUP_DELAY = 'FILTER_GROUP_DELAY',
  FILTER_HIGH_FREQUENCY_HZ = 'FILTER_HIGH_FREQUENCY_HZ',
  FILTER_LOW_FREQUENCY_HZ = 'FILTER_LOW_FREQUENCY_HZ',
  FILTER_PASS_BAND_TYPE = 'FILTER_PASS_BAND_TYPE',
  FILTER_TYPE = 'FILTER_TYPE',

  // Channel steering properties (used in beaming, rotation)
  STEERING_AZIMUTH = 'STEERING_AZIMUTH',
  STEERING_SLOWNESS = 'STEERING_SLOWNESS',

  // Beaming properties
  BEAM_COHERENT = 'BEAM_COHERENT'
}
