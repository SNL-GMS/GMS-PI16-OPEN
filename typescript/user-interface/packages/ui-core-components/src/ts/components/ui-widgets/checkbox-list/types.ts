import { Position } from '@blueprintjs/core';
import { IconName } from '@blueprintjs/icons';
import Immutable from 'immutable';

export interface CheckboxListProps {
  checkboxEnum: any;
  enumToCheckedMap: Map<any, boolean>;
  enumKeysToDisplayStrings?: Map<string, string>;
  enumToColorMap?: Map<any, string>;
  onChange(map: Map<any, boolean>);
}

export interface SimpleCheckboxListProps {
  checkBoxListEntries: CheckboxListEntry[];
  onChange?(name: string): void;
}

export interface SimpleCheckboxListState {
  checkboxEntriesMap: Immutable.Map<string, CheckboxListEntry>;
}

export interface CheckboxListEntry {
  name: string;
  color?: string;
  iconButton?: IconButton;
  isChecked: boolean;
}

export interface IconButton {
  iconName: IconName;
  popover?: {
    position: Position;
    usePortal?: boolean;
    minimal?: boolean;
    content?: string | JSX.Element;
  };
  onClick?(event: React.MouseEvent<HTMLButtonElement>): void;
}
