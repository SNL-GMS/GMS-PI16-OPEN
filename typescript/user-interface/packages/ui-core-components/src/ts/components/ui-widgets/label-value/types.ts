import React from 'react';

/**
 * app determines the default styling for the label-value toolbar item, defaults to soh
 */
export interface LabelValueProps {
  value: string;
  label: string;
  tooltip: string;
  valueColor?: string;
  ianApp?: boolean;
  styleForValue?: React.CSSProperties;
  containerClass?: string;
}
