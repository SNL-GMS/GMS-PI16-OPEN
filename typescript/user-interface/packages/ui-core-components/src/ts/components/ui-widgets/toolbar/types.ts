import { IconName } from '@blueprintjs/core';
import { DateTimeFormat } from '@gms/common-util';
import React from 'react';

import { DateRangePickerTypes } from '../date-range-picker';

export interface ToolbarProps {
  items: ToolbarItem[];
  toolbarWidthPx: number;
  itemsLeft?: ToolbarItem[];
  minWhiteSpacePx?: number;
  spaceBetweenItemsPx?: number;
  hidden?: boolean;
  overflowIcon?: IconName;
}
export interface ToolbarState {
  indicesToOverflow: number[];
  leftIndicesToOverflow: number[];
  whiteSpaceAllotmentPx: number;
  checkSizeOnNextDidMountOrDidUpdate: boolean;
}

export type ToolbarItem =
  | NumericInputItem
  | DropdownItem
  | IntervalPickerItem
  | DateRangePickerItem
  | PopoverItem
  | SwitchItem
  | ButtonItem
  | ButtonGroupItem
  | LabelValueItem
  | CheckboxDropdownItem
  | LoadingSpinnerItem
  | CustomItem;

interface ToolbarItemBase {
  label?: string;
  tooltip: string;
  tooltipForIssue?: string;
  type: ToolbarItemType;
  rank: number;
  style?: React.CSSProperties;
  widthPx?: number;
  labelRight?: string;
  disabled?: boolean;
  icon?: IconName;
  onlyShowIcon?: boolean;
  menuLabel?: string;
  cyData?: string;
  hasIssue?: boolean;
  ianApp?: boolean; // If true, use default ian styling. defaults to false (SoH)
  onMouseEnter?(): void;
  onMouseOut?(): void;
}

interface NumericInputItem extends ToolbarItemBase {
  value: number;
  minMax: MinMax;
  step?: number;
  requireEnterForOnChange?: boolean;
  onChange(value: number);
}

interface DropdownItem extends ToolbarItemBase {
  dropdownOptions: any;
  dropdownText?: any;
  value: any;
  custom?: boolean;
  displayLabel?: boolean;
  onChange(value: any);
}

interface IntervalPickerItem extends ToolbarItemBase {
  startDate: Date;
  endDate: Date;
  defaultIntervalInHours: number;
  shortFormat?: boolean;
  onChange(startDate: Date, endDate: Date);
  onApplyButton(startDate: Date, endDate: Date);
}

interface DateRangePickerItem extends ToolbarItemBase {
  startTimeMs: number;
  endTimeMs: number;
  format: DateTimeFormat;
  durations?: DateRangePickerTypes.DurationOption[];
  minStartTimeMs?: number;
  maxEndTimeMs?: number;
  onChange(startTimeMs: number, endTimeMs: number);
  onApplyButton(startTimeMs: number, endTimeMs: number);
}

interface PopoverItem extends ToolbarItemBase {
  popoverContent: JSX.Element;
  onPopoverDismissed();
}

interface SwitchItem extends ToolbarItemBase {
  value: boolean;
  onChange(value: boolean);
}

interface ButtonItem extends ToolbarItemBase {
  onClick();
}

interface ButtonGroupItem extends ToolbarItemBase {
  buttons: ButtonItem[];
}

interface LabelValueItem extends ToolbarItemBase {
  value: string;
  valueColor?: string;
  styleForValue?: React.CSSProperties;
}

interface CheckboxDropdownItem extends ToolbarItemBase {
  values: Map<any, boolean>;
  colors?: Map<any, string>;
  enumKeysToDisplayStrings?: Map<string, string>;
  enumOfKeys: any;
  onChange(value: any): void;
  onPopUp?(ref?: HTMLDivElement): void;
  onPopoverDismissed?(): void;
}

interface LoadingSpinnerItem extends ToolbarItemBase {
  itemsToLoad: number;
  itemsLoaded?: number;
  hideTheWordLoading?: boolean;
  hideOutstandingCount?: boolean;
}

interface CustomItem extends ToolbarItemBase {
  element: JSX.Element;
}

export interface MinMax {
  min: number;
  max: number;
}

export enum ToolbarItemType {
  Switch = 'Switch',
  Popover = 'Popover',
  Dropdown = 'Dropdown',
  NumericInput = 'NumericInput',
  Button = 'Button',
  IntervalPicker = 'IntervalPicker',
  DateRangePicker = 'DateRangePicker',
  ButtonGroup = 'ButtonGroup',
  LabelValue = 'LabelValue',
  CheckboxList = 'CheckboxList',
  LoadingSpinner = 'LoadingSpinner',
  CustomItem = 'CustomItem'
}

export {
  ButtonGroupItem,
  ButtonItem,
  CheckboxDropdownItem,
  CustomItem,
  DateRangePickerItem,
  DateRangePickerTypes,
  DropdownItem,
  IntervalPickerItem,
  LabelValueItem,
  LoadingSpinnerItem,
  NumericInputItem,
  PopoverItem,
  SwitchItem
};
