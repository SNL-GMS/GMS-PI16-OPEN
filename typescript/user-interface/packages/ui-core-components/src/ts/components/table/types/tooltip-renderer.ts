/* eslint-disable @typescript-eslint/no-empty-interface */
import { ITooltipParams as AgITooltipParams } from 'ag-grid-community';

/** Wrapper interface class around ag-grid interface `ITooltipParams` */
export interface TooltipParams extends AgITooltipParams {}
