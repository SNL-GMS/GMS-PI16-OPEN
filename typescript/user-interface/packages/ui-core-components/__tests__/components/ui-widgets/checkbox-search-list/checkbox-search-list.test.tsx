/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable @typescript-eslint/no-magic-numbers */
import * as Enzyme from 'enzyme';
import * as React from 'react';
import { act } from 'react-dom/test-utils';

import { CheckboxSearchList } from '../../../../src/ts/components/ui-widgets/checkbox-search-list/checkbox-search-list';
import * as CheckboxSearchListTypes from '../../../../src/ts/components/ui-widgets/checkbox-search-list/types';

const TIME_TO_WAIT_MS = 200;

/**
 * Fixes React warning that "An update to Component inside a test was not wrapped in act(...)."
 */
const waitForComponentToPaint = async (wrapper: any): Promise<void> => {
  // eslint-disable-next-line @typescript-eslint/await-thenable
  await act(async () => {
    await new Promise(resolve => setTimeout(resolve, TIME_TO_WAIT_MS));
    wrapper.update();
  });
};

const checkboxItem1 = {
  id: '1',
  name: 'first',
  checked: true
};

const checkboxItem2 = {
  id: '2',
  name: 'second',
  checked: false
};

const checkboxItem3 = {
  id: '3',
  name: 'third',
  checked: false
};

const checkboxItems = [checkboxItem1, checkboxItem2, checkboxItem3];

const props: CheckboxSearchListTypes.CheckboxListProps = {
  items: checkboxItems,
  maxHeightPx: 300,
  onCheckboxChecked: jest.fn()
};

const wrapper = Enzyme.mount(<CheckboxSearchList {...props} />);

describe('CheckboxSearchList', () => {
  it('to be defined', () => {
    expect(CheckboxSearchList).toBeDefined();
  });

  it('CheckboxSearchList shallow renders', () => {
    const shallow = Enzyme.shallow(<CheckboxSearchList {...props} />);
    expect(shallow).toMatchSnapshot();
  });

  it('CheckboxSearchList renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('CheckboxSearchList functions and clicks', async () => {
    const mockOnCheckboxChecked = jest.fn();

    const wrapper3 = Enzyme.mount(
      <CheckboxSearchList
        items={checkboxItems}
        maxHeightPx={300}
        onCheckboxChecked={mockOnCheckboxChecked}
      />
    );

    const cbProps = wrapper3.props() as CheckboxSearchListTypes.CheckboxListProps;
    cbProps.onCheckboxChecked('3', true);
    await waitForComponentToPaint(wrapper3);

    wrapper3.find('.checkbox-search-list__search').simulate('change', { value: 'second' });
    wrapper3.setState({ currentFilter: 'second' });
    await waitForComponentToPaint(wrapper3);
    expect(wrapper3.state('currentFilter')).toEqual('second');

    wrapper3.setState({ currentFilter: '' });
    await waitForComponentToPaint(wrapper3);

    expect(mockOnCheckboxChecked).toHaveBeenCalledTimes(1);
  });

  it('CheckboxSearchList list item clicks', async () => {
    const mockOnCheckboxChecked = jest.fn();

    const wrapper4 = Enzyme.mount(
      <CheckboxSearchList
        items={checkboxItems}
        maxHeightPx={300}
        onCheckboxChecked={mockOnCheckboxChecked}
      />
    );

    const lastCB = wrapper4.find('.checkbox-search-list-item').last();
    const lastInput = lastCB.find('input');
    expect(lastCB).toBeDefined();
    expect(lastInput).toBeDefined();
    lastInput.simulate('change');
    await waitForComponentToPaint(wrapper4);
    expect(mockOnCheckboxChecked).toHaveBeenCalledTimes(1);
  });
});
