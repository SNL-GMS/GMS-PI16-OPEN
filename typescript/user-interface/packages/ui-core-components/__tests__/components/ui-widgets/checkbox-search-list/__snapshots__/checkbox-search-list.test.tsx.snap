// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxSearchList CheckboxSearchList renders 1`] = `
<CheckboxSearchList
  items={
    Array [
      Object {
        "checked": true,
        "id": "1",
        "name": "first",
      },
      Object {
        "checked": false,
        "id": "2",
        "name": "second",
      },
      Object {
        "checked": false,
        "id": "3",
        "name": "third",
      },
    ]
  }
  maxHeightPx={300}
  onCheckboxChecked={[MockFunction]}
>
  <div
    className="checkbox-search-list"
    style={
      Object {
        "height": "300px",
      }
    }
  >
    <input
      autoFocus={true}
      className="checkbox-search-list__search"
      onChange={[Function]}
      placeholder="Search input"
      tabIndex={0}
      type="search"
      value=""
    />
    <div
      className="checkbox-search-list-item"
      key="checkbox-search-list1"
    >
      <Blueprint3.Checkbox
        checked={true}
        onChange={[Function]}
      >
        <Control
          checked={true}
          inputRef={[Function]}
          onChange={[Function]}
          type="checkbox"
          typeClassName="bp3-checkbox"
        >
          <label
            className="bp3-control bp3-checkbox"
          >
            <input
              checked={true}
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="bp3-control-indicator"
            />
          </label>
        </Control>
      </Blueprint3.Checkbox>
      <span
        className="checkbox-search-list-item__label"
      >
        first
      </span>
    </div>
    <div
      className="checkbox-search-list-item"
      key="checkbox-search-list2"
    >
      <Blueprint3.Checkbox
        checked={false}
        onChange={[Function]}
      >
        <Control
          checked={false}
          inputRef={[Function]}
          onChange={[Function]}
          type="checkbox"
          typeClassName="bp3-checkbox"
        >
          <label
            className="bp3-control bp3-checkbox"
          >
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="bp3-control-indicator"
            />
          </label>
        </Control>
      </Blueprint3.Checkbox>
      <span
        className="checkbox-search-list-item__label"
      >
        second
      </span>
    </div>
    <div
      className="checkbox-search-list-item"
      key="checkbox-search-list3"
    >
      <Blueprint3.Checkbox
        checked={false}
        onChange={[Function]}
      >
        <Control
          checked={false}
          inputRef={[Function]}
          onChange={[Function]}
          type="checkbox"
          typeClassName="bp3-checkbox"
        >
          <label
            className="bp3-control bp3-checkbox"
          >
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />
            <span
              className="bp3-control-indicator"
            />
          </label>
        </Control>
      </Blueprint3.Checkbox>
      <span
        className="checkbox-search-list-item__label"
      >
        third
      </span>
    </div>
  </div>
</CheckboxSearchList>
`;

exports[`CheckboxSearchList CheckboxSearchList shallow renders 1`] = `
<div
  className="checkbox-search-list"
  style={
    Object {
      "height": "300px",
    }
  }
>
  <input
    autoFocus={true}
    className="checkbox-search-list__search"
    onChange={[Function]}
    placeholder="Search input"
    tabIndex={0}
    type="search"
    value=""
  />
  <div
    className="checkbox-search-list-item"
    key="checkbox-search-list1"
  >
    <Blueprint3.Checkbox
      checked={true}
      onChange={[Function]}
    />
    <span
      className="checkbox-search-list-item__label"
    >
      first
    </span>
  </div>
  <div
    className="checkbox-search-list-item"
    key="checkbox-search-list2"
  >
    <Blueprint3.Checkbox
      checked={false}
      onChange={[Function]}
    />
    <span
      className="checkbox-search-list-item__label"
    >
      second
    </span>
  </div>
  <div
    className="checkbox-search-list-item"
    key="checkbox-search-list3"
  >
    <Blueprint3.Checkbox
      checked={false}
      onChange={[Function]}
    />
    <span
      className="checkbox-search-list-item__label"
    >
      third
    </span>
  </div>
</div>
`;
